vertica/tests/test_vertica.py

```diff
@@ -85,3 +118,69 @@ def test_custom_queries(aggregator, instance):
     aggregator.assert_metric(
         'vertica.table.force_outer', metric_type=0, tags=['db:datadog', 'foo:bar', 'test:vertica', 'table_name:datadog']
     )
+
+
+@pytest.mark.usefixtures('dd_environment')
+def test_include_all_metric_groups(aggregator, instance):
+    instance['metric_groups'] = [
+        'licenses',
+        'license_audits',
+        'system',
+        'nodes',
+        'projections',
+        'projection_storage',
+        'storage_containers',
+        'host_resources',
+        'query_metrics',
+        'resource_pool_status',
+        'disk_storage',
+        'resource_usage',
+        'version',
+        'custom',
+    ]
+
+    check = VerticaCheck('vertica', {}, [instance])
+    check.check(instance)
+
+    aggregator.assert_metric('vertica.license.expiration', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.license.size', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.node.total', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.projection.total', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.row.total', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.delete_vectors', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.memory.swap.total', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.query.active', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.resource_pool.memory.max', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.storage.size', metric_type=aggregator.GAUGE)
+    aggregator.assert_metric('vertica.node.resource_requests', metric_type=aggregator.GAUGE)
+
+
+@pytest.mark.usefixtures('dd_environment')
+def test_include_system_metric_group(aggregator, instance):
+    instance['metric_groups'] = ['system']
+
+    check = VerticaCheck('vertica', {}, [instance])
+    check.check(instance)
+
+    aggregator.assert_metric('vertica.node.total', metric_type=aggregator.GAUGE)
```

Comment:
Let's assert every metric in `query_licenses()` and `query_system()` then do `aggregator.assert_all_metrics_covered()`